core/src/saros/activities/FileActivity.java

```diff
@@ -77,6 +78,10 @@ public FileActivity(
         break;
     }
 
+    if (content != null && encoding == null) {
+      throw new IllegalArgumentException("Encoding must be passed if binary content is passed");
```

Comment:
Adjust check to require charset if:
- the activity is a file creation
- the activity is a file move that contains binary content.

Requiring the creation of files to always contain a charset ensures that the correct charset is always set, even for files that are created without initial content.

```suggestion
    if (encoding == null && (type == Type.CREATED || (type == Type.MOVED && content != null))) {
      throw new IllegalArgumentException(
          "Encoding must be passed if type is created and/or binary content is passed");
```